starky/src/r1cs2plonk.rs

```diff
@@ -54,13 +69,20 @@ pub fn r1cs2plonk(r1cs: &R1CS<GL>) -> (Vec<PlonkGate>, Vec<PlonkAdd>) {
     let mut plonk_constraints: Vec<PlonkGate> = vec![];
     let mut plonk_additions: Vec<PlonkAdd> = vec![];
 
-    let normalize = |lc: &mut HashMap<usize, FGL>| {
-        lc.retain(|_, v| *v != FGL::ZERO);
+    let normalize = |lc: &mut BTreeMap<usize, FGL>| {
```

Comment:
Actually, the old code here is working too. you can try rollback this change and have a test.  The old code seems more graceful.